------------------------------------------------------------
revno: 14156
revision-id: squid3@treenet.co.nz-20170508110920-73gma737u4x6ce87
parent: squid3@treenet.co.nz-20170504061416-ks61dfut8wyml2qu
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=4695
author: Lubos Uhliarik <luhliari@redhat.com>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Mon 2017-05-08 23:09:20 +1200
message:
  Bug 4695: squidpurge: GCC 7 build errors
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170508110920-73gma737u4x6ce87
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: a0f0c573b5be3d81cf0f8e65ae52bf27bd08dba5
# timestamp: 2017-05-08 11:51:08 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20170504061416-\
#   ks61dfut8wyml2qu
# 
# Begin patch
=== modified file 'tools/purge/purge.cc'
--- tools/purge/purge.cc	2017-01-01 00:16:45 +0000
+++ tools/purge/purge.cc	2017-05-08 11:09:20 +0000
@@ -272,7 +272,7 @@
         snprintf( md5, sizeof(md5), "%-32s", "(no_md5_data_available)" );
     }
 
-    char timeb[64];
+    char timeb[256];
     if ( meta && (findings = meta->search( STORE_META_STD )) ) {
         StoreMetaStd temp;
         // make data aligned, avoid SIGBUS on RISC machines (ARGH!)
@@ -283,7 +283,7 @@
     } else if ( meta && (findings = meta->search( STORE_META_STD_LFS )) ) {
         StoreMetaStdLFS temp;
         // make data aligned, avoid SIGBUS on RISC machines (ARGH!)
-        memcpy( &temp, findings->data, sizeof(StoreMetaStd) );
+        memcpy( &temp, findings->data, sizeof(StoreMetaStdLFS) );
         snprintf( timeb, sizeof(timeb), "%08lx %08lx %08lx %08lx %04x %5hu ",
                   (unsigned long)temp.timestamp, (unsigned long)temp.lastref,
                   (unsigned long)temp.expires, (unsigned long)temp.lastmod, temp.flags, temp.refcount );

